modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogZoneDescriptor.java

```diff
@@ -57,7 +58,7 @@ public class CatalogZoneDescriptor extends CatalogObjectDescriptor implements Ma
      * Returns {@code true} if zone upgrade will lead to assignments recalculation.
      */
     public static boolean updateRequiresAssignmentsRecalculation(CatalogZoneDescriptor oldDescriptor, CatalogZoneDescriptor newDescriptor) {
-        if (oldDescriptor.updateToken() == newDescriptor.updateToken()) {
+        if (oldDescriptor.updateTimestamp() == newDescriptor.updateTimestamp()) {
```

Comment:
Why is it OK to use reference equality here? This might be a bug, **every** reference equality check must be justified with a comment

Reply:
This most likely was missed due to renaming, should be fixed.